### PR TITLE
Skip unit test with optuna version

### PR DESCRIPTION
**Pull request by keisuke-umezawa**
## Contributor License Agreement

This repository (``optuna-dashboard``) and [Goptuna](https://github.com/c-bata/goptuna) share common code.
This pull request may therefore be ported to Goptuna.
Make sure that you understand the consequences concerning licenses and check the box below if you accept the term before creating this pull request.

* [x] I agree this patch may be ported to [Goptuna](https://github.com/c-bata/goptuna) by other Goptuna contributors.

## Reference Issues/PRs

To follow up on https://github.com/optuna/optuna-dashboard/pull/747

## What does this implement/fix? Explain your changes.

- I made unit tests pass with optuna v3.1.0.
- I also checked that unit test passed with v3.2.0 - latest in my local env.
- Added ci with oldest compatible optuna version v3.1.0

### CODE REVIEW

File: python_tests/artifact/test_backend.py [hunk 1]
```diff
@@ -122,6 +128,10 @@ def test_study_artifact_not_found() -> None:
         assert status == 404
 
 
+@pytest.mark.skipif(
+    version.parse(optuna.__version__) < version.parse("3.4.0"),
+    reason="upload_artiract needs storage",
+)
```
Comment (c-bata) [inline python_tests/artifact/test_backend.py:134]: In my understanding, this decorator will be ignored due to the module level `pytest.skip` call in L12. Is that correct?

File: python_tests/test_csv_download.py [hunk 2]
```diff
@@ -5,11 +5,16 @@
...
 from optuna.trial import TrialState
 from optuna_dashboard._app import create_app
+from packaging import version
 import pytest
 
 from .wsgi_client import send_request
 
 
+if version.parse(optuna.__version__) < version.parse("3.4.0"):
+    pytest.skip("Study.metrics_name is introduced at v3.4.0", allow_module_level=True)
```
Comment (c-bata) [inline python_tests/test_csv_download.py:15]: [fyi] This dependency should be fixed, but it looks out of the scope of this PR.

https://github.com/optuna/optuna-dashboard/pull/723/files#diff-0f7b622fa29613e91a846a5635ff2339348136638848391f330189b81b224a80R471-R472

File: python_tests/artifact/test_backend.py (hunk 1 above)
Comment (keisuke-umezawa) [reply python_tests/artifact/test_backend.py:134]:
```
if version.parse(optuna.__version__) < version.parse("3.3.0"):
    pytest.skip("optuna.artiracts module is introduced at v3.3.0", allow_module_level=True)
```

At the module level, we will skip tests < v3.3.0. But, only this test does not pass at v3.3.0. So, I added skipif only for this test.

Comment (c-bata) [reply python_tests/artifact/test_backend.py:134]: Ah, I understand. Thank you for your reply :pray:

**Review by c-bata — APPROVED** [review]
LGTM!

**Pull request merged**